bundles/org.openhab.binding.awattar/src/main/java/org/openhab/binding/awattar/internal/handler/aWATTarPriceHandler.java

```diff
@@ -0,0 +1,186 @@
+import static org.openhab.binding.awattar.internal.aWATTarBindingConstants.CHANNEL_MARKET_NET;
+import static org.openhab.binding.awattar.internal.aWATTarBindingConstants.CHANNEL_TOTAL_GROSS;
+import static org.openhab.binding.awattar.internal.aWATTarBindingConstants.CHANNEL_TOTAL_NET;
+import static org.openhab.binding.awattar.internal.aWATTarUtil.getCalendarForHour;
+import static org.openhab.binding.awattar.internal.aWATTarUtil.getMillisToNextMinute;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.time.ZonedDateTime;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.awattar.internal.aWATTarPrice;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.type.ChannelKind;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link aWATTarPriceHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Wolfgang Klimt - Initial contribution
+ */
+@NonNullByDefault
+public class aWATTarPriceHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(aWATTarPriceHandler.class);
+
+    private int thingRefreshInterval = 60;
+    private @Nullable ScheduledFuture<?> thingRefresher;
+
+    public aWATTarPriceHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.trace("Handling command {} for channel {}", command, channelUID);
+        if (command instanceof RefreshType) {
+            refreshChannel(channelUID);
+        } else {
+            logger.debug("Binding {} only supports refresh command", BINDING_ID);
+        }
+    }
+
+    /**
+     * Initialize the binding and start the refresh job.
+     * The refresh job runs once after initialization and afterwards every hour.
+     */
+
+    @Override
+    public void initialize() {
+        logger.trace("Initializing aWATTar price handler {}", this);
+
+        synchronized (this) {
+            ScheduledFuture<?> localRefresher = thingRefresher;
+            if (localRefresher == null || localRefresher.isCancelled()) {
+                logger.trace("Start Thing refresh job at interval {} seconds.", thingRefreshInterval);
+                thingRefresher = scheduler.scheduleAtFixedRate(this::refreshChannels, getMillisToNextMinute(1),
```

Comment:
See above. Please check below in this file.

Reply:
This one is still necessary to meet the minute boundaries.